### PR TITLE
Added Configuration and basic trello list cards

### PR DESCRIPTION
**Pull request by mstead**
A Trilla class encapsulates the application controller
logic. The CLI should do its work through that class.

Added a config framework that gets passed along. Config should
be in the ~/.trilla/trilla.yaml file.

On command execution, a config class is instantiated
and passed through click's excution context. Any required
configuration changes via command line options should be
done along the command chain, and before the Trilla
object is created.

A basic Trello connection class was added to wrap py-trello
and will provide the trello API calls that we will require.

Added a 'trilla track cards' command to test out the trello connection.
Can remove it if it isn't needed.

### CODE REVIEW

File: requirements_dev.txt [hunk 1]
```diff
@@ -9,3 +9,4 @@ Sphinx==1.4.4
 cryptography==1.4
 PyYAML==3.11
 pytest==2.9.2
+py-trello==0.5.0
```
Comment (vritant) [inline requirements_dev.txt]: I might be wrong, but this should be in setup.py.
requirements_dev.txt I think is for packages that help us to develop and are not needed for shipping trilla itself.

**Comment by vritant** [comment]
@mstead im good with merge except that one comment about the dependency.

File: requirements_dev.txt (hunk 1 above)
Comment (mstead) [inline requirements_dev.txt]: > I might be wrong, but this should be in setup.py.
> requirements_dev.txt I think is for packages that help us to develop and are not needed for shipping trilla itself.

Fixed and committed.

**Comment by mstead** [comment]
@vritant I updated the dep issue yesterday. Good catch, had it in the wrong place.

**Pull request merged**